pkg/pillar/devicenetwork/dns.go

```diff
@@ -150,21 +154,27 @@ func ResolveWithPortsLambda(domain string,
 						errs = append(errs, err)
```

Comment:
we can send errors through channel to avoid mutex

Reply:
not sure if that is a gain, because then in the end I have to start another goroutine to receive them and append them to the array, or make the channel big enough to store them in there and then convert them to an array

Reply:
We could also use [sync.map[(https://pkg.go.dev/sync#Map) but I think Mx is fine for this case.